editor/src/components/canvas/canvas-utils.ts

```diff
@@ -446,36 +447,39 @@ export function updateFramesOfScenesAndComponents(
             if (parentElement == null) {
               throw new Error(`Unexpected result when looking for parent: ${parentElement}`)
             }
-            // Flex based layout.
-            const possibleFlexProps = FlexLayoutHelpers.convertWidthHeightToFlex(
-              frameAndTarget.newSize.width,
-              frameAndTarget.newSize.height,
-              element.props,
-              right(parentElement.props),
-              eitherToMaybe(FlexLayoutHelpers.getMainAxis(right(parentElement.props))),
-              frameAndTarget.edgePosition,
+
+            const currentAttributeToChange =
+              eitherToMaybe(
+                getSimpleAttributeAtPath(
+                  right(element.props),
+                  createLayoutPropertyPath(frameAndTarget.targetProperty),
+                ),
+              ) ?? 0
+
+            const newAttributeValue = jsxAttributeValue(
+              currentAttributeToChange + frameAndTarget.delta,
+              emptyComments,
             )
-            forEachRight(possibleFlexProps, (flexProps) => {
-              const { flexBasis, width, height } = flexProps
-              if (flexBasis != null) {
-                propsToSet.push({
-                  path: createLayoutPropertyPath('flexBasis'),
-                  value: jsxAttributeValue(flexBasis, emptyComments),
-                })
-              }
-              if (width != null) {
-                propsToSet.push({
-                  path: createLayoutPropertyPath('Width'),
-                  value: jsxAttributeValue(width, emptyComments),
-                })
-              }
-              if (height != null) {
-                propsToSet.push({
-                  path: createLayoutPropertyPath('Height'),
-                  value: jsxAttributeValue(height, emptyComments),
-                })
-              }
+
+            propsToSet.push({
+              path: createLayoutPropertyPath(frameAndTarget.targetProperty),
+              value: newAttributeValue,
             })
+
+            propsToSkip.push(
```

Comment:
Does this need to exclude the prop that is being set (`frameAndTarget.targetProperty`)?

Reply:
Shouldn't do no, that's not how the logic for this whole function works.